changelog_unreleased/javascript/pr-8016.md

```diff
@@ -0,0 +1,16 @@
+#### Keep curly braces and `from` keyword in empty `import` statements ([#8016](https://github.com/prettier/prettier/pull/8016) by [@fisker](https://github.com/fisker))
+
+<!-- prettier-ignore -->
+```js
+// Input
+import { } from 'foo';
+import { /* comment */ } from 'bar';
+
+// Prettier stable
+import {} from 'foo'
+import /* comment */ 'bar'
+
+// Prettier master
+import {} from "foo";
+import {} from /* comment */ "bar";
```

Comment:
This is not the expected output in https://github.com/prettier/prettier/issues/7912

Reply:
Don't know how to fix this, similar one

**Prettier 2.0.4**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzCAnABACgDpRZYD0AVJALYUIynFYDcBRZl1sdThJ5EVNdAgEoswAL4gANCAgAHGAEtoAZ2SgAhhgwQA7gAVNCVSnUAbHeoCeq6QCMM6sAGs4MAMqzHCqAHNkMDABXOGkACxgKUwB1UIV4ZU8wODcjOIUANzjLZHBlGxBvZTgMGD0HHwp1ZFQzIukAK2UADwAhB2dXN3VqABlvOGrakJBGprdvH1M4AEVAiHhB0zqQTwwijBzbdVs4UykVjG8YKIUAExhQ5AAOAAZpWW0iqIdZHIe4dfSB6QBHOfgynJjCB1MoALRQOBwU7Q-YYOB-BTwsrqCpVJA1JbDIoUBT+ILYiZTWbzAYYobSGDbE7nS5IABMlIcClMEwAwnxKjkPgBWfaBIoAFW2xkxy3SwQAklAYbA3GBDvIAIIytwwSxTRZFMRiIA)
```sh
--parser babel
```

**Input:**
```jsx
for (
  /*comment*/ ;
  /*comment*/;
  /*comment*/
) {}
```

**Output:**
```jsx
for (;;) /*comment*/ /*comment*/ /*comment*/
{}

```

**Second Output:**
```jsx
for (;;) /*comment*/ /*comment*/ /*comment*/ {}

```

Reply:
You can first add these comments as "dangling" (if `node.specifiers.length === 0 && next significant character === '}'`) and then print the dangling comments inside the curly braces.